src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx

```diff
@@ -442,19 +455,49 @@ const TableBodyRow = ({
             align={align}
             className={`data-table-${align}`}
           >
-            <StyledNestedSpan nestingOffset={offset}>
-              {col.renderDataFunction ? (
-                col.renderDataFunction({
-                  // Call the column renderer if it's provided
-                  value: values[col.columnId],
-                  columnId: col.columnId,
-                  rowId: id,
-                  row: values,
-                })
-              ) : (
-                <TableCellRenderer>{values[col.columnId]}</TableCellRenderer>
-              )}
-            </StyledNestedSpan>
+            {activeBar !== 'edit' ? (
+              <StyledNestedSpan nestingOffset={offset}>
+                {col.renderDataFunction ? (
+                  col.renderDataFunction({
+                    // Call the column renderer if it's provided
+                    value: values[col.columnId],
+                    columnId: col.columnId,
+                    rowId: id,
+                    row: values,
+                  })
+                ) : (
+                  <TableCellRenderer>{values[col.columnId]}</TableCellRenderer>
+                )}
+              </StyledNestedSpan>
+            ) : (
+              <StyledNestedSpan nestingOffset={offset}>
+                {col.editDataFunction ? (
+                  <CarbonTextInput
```

Comment:
No, we cannot just use a CarbonTextInput in all cases.  This is why when we talked we said that an `editDataFunction` should be like the `renderDataFunction` and when there is an `editDataFunction` then we use that.  Otherwise you can default to using carbon text input if you want, but, we need to be able to supply our own editor.  eg, booleans will use a checkbox/switch, dates should have a picker, etc.  If the table isn't going to manage all the possible types, etc, then it just needs to delegate to something that will.

Reply:
Yeah, the original discussion is here https://github.ibm.com/Watson-IoT/pal-tracking/issues/20

I agree and think it's clearer to not provide any `Input` by default, but instead rely on the `editDataFunction` entirely. Then, in the story we can show how to use this with different input types.

Reply:
Yes, I understood that at first the time. But I thought we could supply our own editor in `TableBodyRow` not in `Table.story`, besides `CarbonTextInput` I wrote is just a demo. But now, I move the contents of editDataFunction into Table.story, besides user can customize the contents of editDataFunction - I have wrote two functions for demo, the one is `editInputData` which will render `<TextInput>`, the other is `editSelectData` which will render `<Select>`. User can alse add some checkbox/switch, picker, etc. they want.